### ISSUE TITLE
TOML Java library differentiation

### ISSUE THREAD

**Issue by calexHG** [TheElectronWill/TOML-javalib#4]
Good ~~morning~~ afternoon Guillaume! 

My JVM app in development is heavily considering using the TOML format. I have noticed at least one other Java library in addition to yours: [toml4j](https://github.com/mwanji/toml4j). Maybe there are also others.

I assume you are very familiar with toml4j? It appears that toml4j has been in development since 2013, whereas TOML-javalib is new as of this year. Can any implications be made from that? What differentiates your library? Do you have different quality standards..  api design patterns.. different goals? 

Note that this may be a common question that many other potential users of your graciously free library may have. I sincerely appreciate any time you spend towards giving as thorough an answer as possible. 

Salut!

**Comment by TheElectronWill** [comment]
That's a good question! I have already used toml4j, and I think that the main difference between my library and toml4j is that I focus on **simplicity**. In my opinion, an "obvious, minimal language" like TOML shouldn't require a complicated library but a very straightforward one. And that's what I try to do.

Here are some noticeable differences between the two libraries:
- The source code of TOML-javalib is simpler and shorter than the one of toml4j. Therefore it's smaller in size.
- I don't care too much about thread-safety whereas toml4j tries to be thread-safe.
- TOML-javalib uses java 8 but toml4j supports java 6.
- TOML-javalib has no specific getter like `getString`, `getInt`, etc. because `Toml.read` just returns an instance of `Map<String, Object>`. In fact, I designed my library to directly use the Java's classes.
- TOML-javalib doesn't support the complex feature "custom classes" (might be added later if it's really useful) which maps toml data to a java class.
- TOML-javalib has **no dependancy**, unlike toml4j.
- TOML-javalib distinguishes 3 kinds of datetimes: ZonedDateTime, LocalDateTime and LocalDate, based on what informations are available.
- Because I also care about the user who writes the toml file, I added support for "lenient bare keys". It's an option that allows bare keys to contain almost any utf character. Of course, this option doesn't strictly follow the current TOML specification, but it's so practical, especially for non-english users!
